vmbindings/dummyvm/src/tests/barrier_slow_path_assertion.rs

```diff
@@ -0,0 +1,30 @@
+// GITHUB-CI: MMTK_PLAN=GenImmix
+// GITHUB-CI: FEATURES=vo_bit, extreme_assertions
+use crate::object_model::OBJECT_REF_OFFSET;
+use crate::{api::*, edges};
+use atomic::Atomic;
+use mmtk::util::{Address, ObjectReference, VMMutatorThread, VMThread};
+use mmtk::vm::edge_shape::SimpleEdge;
+use mmtk::AllocationSemantics;
+
+
+#[test]
+#[should_panic]
+fn test_assertion_barrier() {
```

Comment:
Is this test supposed to panic? Seems like the barrier is getting called on `objref` which is a valid object. Maybe try something like `objref + 0x4` or something?

Reply:
The test should panic(and it panic on my local machine)because the `post_alloc` is not called for the object, which makes the object's metadata remain unitialized. So the assertion `debug_assert!(is_vo_bit_set::<S::VM>(object), "object bit is unset");` fails.

Reply:
I think the test will panic. It did not panic in the CI tests, as the CI script did not extract the features correctly, and ran the test without any feature.

However, I still find the test a bit hacky. We expect the users to call `post_alloc` for every object they allocate. This test deliberately does not call `post_alloc` to create a scenario that the alloc bit is not set. We could just create an object normally (with `alloc` and `post_alloc`), then create an internal reference and use that as the object reference, which of course is not a valid object reference and its alloc bit is not set. In addition to this, we may also want a test case that passes.

I will push some changes to the PR.